types/module/module.go

```diff
@@ -50,7 +51,7 @@ type AppModuleBasic interface {
 	RegisterLegacyAminoCodec(*codec.LegacyAmino)
 	RegisterInterfaces(codectypes.InterfaceRegistry)
 
-	DefaultGenesis(codec.JSONCodec) json.RawMessage
+	DefaultGenesis() proto.Message
 	ValidateGenesis(codec.JSONCodec, client.TxEncodingConfig, json.RawMessage) error
```

Comment:
Should we replace the json in `ValidateGenesis`? If yes, let's do it in a separate PR; this one is already too big

Reply:
Agreed.